api/src/main/java/edu/cornell/mannlib/vitro/webapp/controller/freemarker/DeleteIndividualController.java

```diff
@@ -0,0 +1,200 @@
+    byte[] toRemove = getIndividualsToDelete(individualUri, delteQueryText, vreq);
+    if (toRemove.length > 0) {
+      deleteIndividuals(toRemove, vreq);
+    }
+    String redirectUrl = getRedirectUrl(vreq);
+
+    return new RedirectResponseValues(redirectUrl, HttpServletResponse.SC_SEE_OTHER);
+  }
+	
+  private String getRedirectUrl(VitroRequest vreq) {
+    String redirectUrl = vreq.getParameter("redirectUrl");
+    if (redirectUrl != null) {
+      return redirectUrl;
+    }
+    return "/";
+  }
+
+
+  private TemplateResponseValues prepareErrorMessage(String errorMessage) {
+    HashMap<String, Object> map = new HashMap<String, Object>();
+    map.put("errorMessage", errorMessage);
+    return new TemplateResponseValues("error-message.ftl", map);
+  }
+	
+  private String handleErrors(VitroRequest vreq) {
+    String uri = vreq.getParameter("individualUri");
+    if (uri == null) {
+      return "Individual uri is null. No object to delete.";
+    }
+    if (uri.contains(">")) {
+      return "Individual uri shouldn't contain >";
+    }
+    return "";
+  }
+
+  private static String getDeleteQueryForType(String typeURI, Model displayModel) {
+    String deleteQueryText = DEFAULT_DELETE_QUERY_TEXT;
+    Query queryForTypeSpecificDeleteQuery = QueryFactory.create(queryForDeleteQuery);
+    QuerySolutionMap initialBindings = new QuerySolutionMap();
+    initialBindings.add("associatedURI", ResourceFactory.createResource(typeURI));
+    displayModel.enterCriticalSection(Lock.READ);
+    try {
+      QueryExecution qexec = QueryExecutionFactory.create(queryForTypeSpecificDeleteQuery, displayModel, initialBindings);
+      try {
+        ResultSet results = qexec.execSelect();
+        while (results.hasNext()) {
+          QuerySolution solution = results.nextSolution();
+          deleteQueryText = solution.get("deleteQueryText").toString();
+        }
+      } finally {
+        qexec.close();
+      }
+    } finally {
+      displayModel.leaveCriticalSection();
+    }
+
+    if (!deleteQueryText.equals(DEFAULT_DELETE_QUERY_TEXT)) {
+      log.debug("For " + typeURI + " found delete query \n" + deleteQueryText);
+    } else {
+      log.debug("For " + typeURI + " delete query not found. Using defalut query \n" + deleteQueryText);
+    }
+    return deleteQueryText;
+  }
+	
+  private String getObjectMostSpecificType(String individualURI, VitroRequest vreq) {
+    String type = "";
+    try {
+      ResultSet results = QueryUtils.getLanguageNeutralQueryResults(makeTypeQuery(individualURI), vreq);
+      while (results.hasNext()) {
+        QuerySolution solution = results.nextSolution();
+        type = solution.get("type").toString();
+        log.debug(type);
+      }
+    } catch (Exception e) {
+      log.error("Failed to get type for individual URI " + individualURI);
+      log.error(e, e);
+    }
+    return type;
+  }
+  
+  private byte[] getIndividualsToDelete(String targetIndividual, String deleteQuery, VitroRequest vreq) {
```

Comment:
I am not sure the best return value of this function is byte[]. Can you consider encapsulating conversion of Model to byte[] and creation of InputStream within the deleteIndividuals method?

Reply:
Sure